src/overlays/actors/ovl_Bg_Ikana_Dharma/z_bg_ikana_dharma.c

```diff
@@ -28,47 +34,221 @@ const ActorInit Bg_Ikana_Dharma_InitVars = {
     (ActorFunc)BgIkanaDharma_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_80BECAD0 = {
-    { COLTYPE_NONE, AT_NONE, AC_ON | AC_TYPE_PLAYER, OC1_NONE, OC2_NONE, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK0, { 0x00000000, 0x00, 0x00 }, { 0x00000100, 0x00, 0x00 }, TOUCH_NONE | TOUCH_SFX_NORMAL, BUMP_ON, OCELEM_NONE, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_NONE,
+        OC2_NONE,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000100, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_NONE,
+    },
     { 98, 10, 25, { 0, 0, 0 } },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80BECAFC[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 320, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 320, ICHAIN_CONTINUE),
     ICHAIN_F32_DIV1000(gravity, -1100, ICHAIN_STOP),
 };
 
-#endif
-
-extern ColliderCylinderInit D_80BECAD0;
-extern InitChainEntry D_80BECAFC[];
-
-extern UNK_TYPE D_060008C8;
-extern UNK_TYPE D_06000C50;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC240.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/BgIkanaDharma_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/BgIkanaDharma_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC5C4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC5E0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC758.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC790.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC7EC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/func_80BEC808.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/BgIkanaDharma_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Ikana_Dharma/BgIkanaDharma_Draw.s")
+static BgIkanaDharma* sFirstHitBgIkanaDharma;
+
+void BgIkanaDharma_CreateParticles(BgIkanaDharma* this, GlobalContext* globalCtx) {
+    u32 pad;
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f accel;
+    f32 speed;
+    f32 dirX;
+    f32 initialY;
+    f32 dirZ;
+    f32 initialRadius;
+    s16 angle;
+    s32 i;
```

Comment:
Only `angle` and `dirZ`.

`this->dyna.actor.scale.y` won't be hoisted outside the loop by the compiler, since `EffectSsKiraKira_SpawnSmallYellow` might change it (as far as the compiler knows).

And moving anything before `initialY` messes up the order.

One thing that can be done: initialize `initialRadius` in the declaration.

Also do you have a preference for "initial position" vs just "position"? It is the position where the particle is created; particles move, obviously

Reply:
`position` (or even `pos`) is fine: once that function's called, this actor stops worrying about it.